### ISSUE TITLE
Large paths stay out of sync w/ Android version

### ISSUE THREAD

**Issue by mshopf** [syncthing/syncthing#2153]
I have the Android (5.0) version of syncthing (v0.11.13, gui v0.6.5) syncing /sdcard (about 6GB) to my Linux system (v0.11.19, 64bit). I have only these two machines, and only a single path configured. There is no .stignore on the Android device yet.
Syncing started fine and at good speed, but at some point of time syncing just stopped, and 3200 items stay out-of-sync (global 4167 items, local 967). The PC itself stays on "Syncing" for the Android device, the Android system says "Up to Date" for the PC device. When I look at the files that ought to be out-of-sync, I cannot find a certain pattern, or a special file (e.g. with wrong rights). But the list of out-of-sync files isn't exactly useful, as no reason and even no path specification is listed.
The log doesn't tell me anything reasonable, it also doesn't contain date stamps, and apparently logging stopped after self-update, so there's nothing about the latest run. Unfortunately, the Android device doesn't seem to be able to create a log file (open '/dev/hwlog_switch' fail -1,13. Permission devnied).
After some time (a few days) when the two systems couldn't communicate, some(!) files transferred, but not all.
I'm lost how to debug this, especially I don't have any experience with go, which information would be needed? Is this related to #2148? I doubt, though.

**Comment by AudriusButkevicius** [comment]
Enable STTRACE=model or set that via android debug options, which should make the log more verbose. Also I'd verify that the states you are seeing are showing the same way in the web ui (rather than app). Also, the tooltips (after you hover on a file) shows full path and potentially date

**Comment by mshopf** [comment]
The Web GUI show definitely some differences (Native: "host: Syncing (99%)", Web: "host (74%)" - the later is more likely), but the out-of-sync items are on the PC side (where there is no native GUI). There's no up- nor download going on. Thanks for the tip with the tooltips, I'll check some of the paths. 
Enabling model debugging on the PC increases verbosity (I shortened the log, and the phone device id to XYZ):
```
[monitor] 2015/08/13 13:20:12.663852 monitor.go:94: INFO: Starting syncthing
[D2BHN] 2015/08/13 13:20:13.349156 main.go:473: INFO: syncthing v0.11.19 (go1.4.2 linux-amd64 default) unknown-user@syncthing-builder 2015-08-09 09:56:22 UTC
[...]
[D2BHN] 2015/08/13 13:20:13.498497 main.go:760: INFO: Database block cache capacity 29656 KiB
[D2BHN] 2015/08/13 13:20:13.987300 progressemitter.go:99: DEBUG: progress emitter: updated interval 5s
[D2BHN] 2015/08/13 13:20:13.988522 progressemitter.go:61: DEBUG: progress emitter: timer - looking after 0
[D2BHN] 2015/08/13 13:20:13.988673 progressemitter.go:77: DEBUG: progress emitter: nothing new
[...]
[D2BHN] 2015/08/13 13:20:15.794670 connections.go:176: INFO: Established secure connection to XYZ at 192.168.3.3:22000-192.168.3.31:32799
[D2BHN] 2015/08/13 13:20:15.795858 model.go:998: DEBUG: sendIndexes for XYZ-192.168.3.3:22000-192.168.3.31:32799/"phone" starting
[D2BHN] 2015/08/13 13:20:15.811584 model.go:1060: DEBUG: sendIndexes for XYZ-192.168.3.3:22000-192.168.3.31:32799/"phone": 314 files (<273020 bytes) (initial index)
[D2BHN] 2015/08/13 13:20:15.877178 model.go:597: INFO: Device XYZ client is "syncthing v0.11.13"
[D2BHN] 2015/08/13 13:20:15.877635 model.go:602: INFO: Device XYZ name is "p8"
[D2BHN] 2015/08/13 13:20:15.972695 model.go:1068: DEBUG: sendIndexes for XYZ-192.168.3.3:22000-192.168.3.31:32799/"phone": 12 files (<257320 bytes) (batched update)
[D2BHN] 2015/08/13 13:20:15.982844 model.go:485: DEBUG: IDX(in): XYZ "phone": 807 files
[D2BHN] 2015/08/13 13:20:16.222364 model.go:1068: DEBUG: sendIndexes for XYZ-192.168.3.3:22000-192.168.3.31:32799/"phone": 158 files (<256660 bytes) (batched update)
[D2BHN] 2015/08/13 13:20:16.289994 model.go:533: DEBUG: model@0xc20879e7e0 IDXUP(in): XYZ / "phone": 880 files
[D2BHN] 2015/08/13 13:20:16.309996 progressemitter.go:145: DEBUG: progress emitter: bytes completed for phone: 0
[D2BHN] 2015/08/13 13:20:16.310187 model.go:404: DEBUG: model@0xc20879e7e0 NeedSize("phone"): 235 30771926
[D2BHN] 2015/08/13 13:20:16.310476 rwfolder.go:189: DEBUG: rwFolder/phone@0xc2081b9c20 remote index updated, rescheduling pull
```
The sendIndexes/IDXUP/progressemitter/NeedSize/reschedule cycle continues several times with different file numbers. At the end I get
```
[D2BHN] 2015/08/13 13:20:17.721940 model.go:404: DEBUG: model@0xc20879e7e0 NeedSize("phone"): 3199 585341043
[D2BHN] 2015/08/13 13:20:17.891021 model.go:327: DEBUG: model@0xc20879e7e0 Completion(XYZ, "phone"): 99.999649 (20290 / 5779307318)
[D2BHN] 2015/08/13 13:20:18.657791 rwfolder.go:226: DEBUG: rwFolder/phone@0xc2081b9c20 pulling 0 6832
[D2BHN] 2015/08/13 13:20:18.657961 rwfolder.go:393: DEBUG: rwFolder/phone@0xc2081b9c20 c 1 p 16
[D2BHN] 2015/08/13 13:20:20.158854 rwfolder.go:238: DEBUG: rwFolder/phone@0xc2081b9c20 changed 0
[D2BHN] 2015/08/13 13:20:20.158910 rwfolder.go:260: DEBUG: rwFolder/phone@0xc2081b9c20 next pull in 10s
[D2BHN] 2015/08/13 13:20:20.468390 progressemitter.go:145: DEBUG: progress emitter: bytes completed for phone: 0
[D2BHN] 2015/08/13 13:20:20.468511 model.go:404: DEBUG: model@0xc20879e7e0 NeedSize("phone"): 3201 1483391727
[D2BHN] 2015/08/13 13:20:20.656065 model.go:327: DEBUG: model@0xc20879e7e0 Completion(XYZ, "phone"): 99.999649 (20290 / 5779307318)
[D2BHN] 2015/08/13 13:20:30.159509 rwfolder.go:219: DEBUG: rwFolder/phone@0xc2081b9c20 skip (curVer == prevVer) 6832 true
```
The 'skip' line repeats 6 times, once every 10 seconds. Then comes a rescan again.

Enabling model debugging on the Android system apparently doesn't help anything:
```
I/SyncthingService(30371): Web GUI will be available at https://127.0.0.1:8384
I/SyncthingService(30371): Starting syncthing according to current state and preferences
I/SyncthingService(30371): Web GUI has come online at https://127.0.0.1:8384
```
... and that's it (adb logcat PollWebGuiAvailableTask:V SyncthingService:V PostTask:V "*:S").

Does this already help anything?

**Comment by mshopf** [comment]
After I deleted some files on the PC side, the Android Web GUI (only that) showed some conflicts as well, and an "Override" button - I actually set the phone as folder master in order to not loose any data before I have everything going. When I pressed it, the deleted files were sent to the PC side again (as expected). Note that the button was not there before I deleted the files on the PC.
After that action, transfer started again, and in almost no time the remaining GBs had been transferred.

This is weird, apparently syncthing just waited for some event just to start working again. What can I do to help debug this issue? It seems to be very intermediate.

**Comment by AudriusButkevicius** [comment]
Well if you can provide exact steps of reproduction that would be great. But I think the issue lies in Android's inotify implementation, (hence closing and you should reopen there) as it just failed to notify us about new files/changes. I bet manual rescan would have resolved the same issue.

**Comment by mshopf** [comment]
No, the state shows clearly, that the items are out-of-sync, so syncthing *knows* that they shall be transferred. Resync didn't help, tried that a million times. I also reconstructed the database, both on PC and Android, to no avail. I removed the path config and recreated it, it didn't help either.
I'll eventually try to reproduce, but my guess is that this is difficult. If I can reproduce, I'd try to reproduce on a pure PC<>PC sync.

I personally don't think the issue is closed, but I guess you're right, nobody can help ATM as long as this is not reproducible...
